src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java

```diff
@@ -327,6 +327,7 @@ public void supplyVanillaManipulators(List<DataManipulator<?, ?>> manipulators)
         if (this.fire > 0) {
             manipulators.add(get(IgniteableData.class).get());
         }
+        manipulators.add(new SpongeGravityData());
```

Comment:
No no, you need to pass in the entity's current gravity state, from `hasNoGravity`.